File: 08-Greedy-Algorithm/java/01-NMeetingsInOneRoom.java
class Solution {
    public int maxMeetings(int[] start, int[] end) {
        int n = start.length;

        Integer[] idx = new Integer[n];
        for (int i = 0; i < n; i++) {
            idx[i] = i;
        }

        Arrays.sort(idx, (a, b) -> {
            if (end[a] == end[b]) {
                return start[a] - start[b];
            }
            return end[a] - end[b];
        });

        int count = 1;
        int lastEnd = end[idx[0]];

        for (int i = 1; i < n; i++) {
            if (start[idx[i]] > lastEnd) {
                count++;
                lastEnd = end[idx[i]];
            }
        }

        return count;
    }
}

File: 08-Greedy-Algorithm/python/01_n_meetings_in_one_room.py
class Solution:
    def maxMeetings(self, start: List[int], end: List[int]) -> int:
        n = len(start)
        meetings = [(start[i], end[i]) for i in range(n)]
        meetings.sort(key=lambda x: (x[1], x[0]))

        count = 1
        last_end = meetings[0][1]

        for i in range(1, n):
            if meetings[i][0] > last_end:
                count += 1
                last_end = meetings[i][1]

        return count
